pkgs/development/compilers/llvm/11/clang/default.nix

```diff
@@ -19,26 +20,31 @@ let
       mv clang-tools-extra-* $sourceRoot/tools/extra
     '';
 
-    nativeBuildInputs = [ cmake python3 lld ]
+    nativeBuildInputs = [ cmake python3 ]
       ++ lib.optional enableManpages python3.pkgs.sphinx
       ++ lib.optional stdenv.hostPlatform.isDarwin fixDarwinDylibNames;
 
-    buildInputs = [ libxml2 llvm ];
+    buildInputs = [ libxml2 llvm.dev lld ];
```

Comment:
there is `lib.getDev` mapped on these by default, so `lld is fine. Just need `llvm.dev` to work around the explicit output as described above.

Also `lld` is removed because I don't think it's needed, and `llvm` is no` libllvm` obviating the issue.